### PR TITLE
BAN-2158 Add text boxes to Apple News

### PR DESCRIPTION
**Pull request by damiendillon**
Create text box elements that:
- contain headings, lists and text
- have a distinguishable style and layout from headings, lists and text 
which are not in text-boxes

https://mydevex.atlassian.net/browse/BAN-2158

### CODE REVIEW

**Comment by ozgun** [comment]
+1 Looks good!

**Comment by EdLittle** [comment]
+1

File: lib/article_json/export/apple_news/elements/text_box.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            {
+              role: 'container',
+              layout: 'textBoxLayout',
+              style: 'textBoxStyle',
+              animation: {
+                type: 'appear',
+                userControllable: true,
+                initialAlpha: 0.0,
+              },
+              components: map_styles(elements)
```
Comment (dsager) [inline lib/article_json/export/apple_news/elements/text_box.rb]:
```suggestion
              components: map_styles(elements),
```

File: lib/article_json/export/apple_news/elements/text_box.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+              klass = child_element.class
+              if klass == ArticleJSON::Elements::Heading
+                namespace::Heading.new(child_element).export
+              elsif klass == ArticleJSON::Elements::Paragraph
+                namespace::Paragraph.new(child_element).export
+              elsif klass == ArticleJSON::Elements::List
+                namespace::List.new(child_element).export
+              else
+                namespace::Text.new(child_element).export
+              end
```
Comment (dsager) [inline lib/article_json/export/apple_news/elements/text_box.rb]:
```suggestion
              case klass
              when ArticleJSON::Elements::Heading
                namespace::Heading.new(child_element).export
              when ArticleJSON::Elements::Paragraph
                namespace::Paragraph.new(child_element).export
              when ArticleJSON::Elements::List
                namespace::List.new(child_element).export
              else
                namespace::Text.new(child_element).export
              end
```

File: spec/article_json/export/apple_news/elements/text_box_spec.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
+describe ArticleJSON::Export::AppleNews::Elements::TextBox do
+  subject(:element) { described_class.new(source_element) }
+  let(:float) { nil }
+  let(:output) do
+    {
+      animation: {initialAlpha: 0.0, type: 'appear', userControllable: true},
```
Comment (dsager) [inline spec/article_json/export/apple_news/elements/text_box_spec.rb]:
```suggestion
      animation: { initialAlpha: 0.0, type: 'appear', userControllable: true },
```

File: lib/article_json/export/apple_news/elements/text_box.rb (hunk 1 above)
Comment (damiendillon) [reply lib/article_json/export/apple_news/elements/text_box.rb]: Amened. Thanks.

File: spec/article_json/export/apple_news/elements/text_box_spec.rb (hunk 3 above)
Comment (damiendillon) [reply spec/article_json/export/apple_news/elements/text_box_spec.rb]: Amended, thanks.

File: lib/article_json/export/apple_news/elements/text_box.rb (hunk 2 above)
Comment (damiendillon) [reply lib/article_json/export/apple_news/elements/text_box.rb]: Through a quick pairing we noted that `if` and `case` statements are not an exact like for like if you use `==` like I did here. `case` statements actually use `===` for comparison.

**Pull request merged**